api4/user_test.go

```diff
@@ -2068,8 +2069,19 @@ func TestUserLoginMFAFlow(t *testing.T) {
 	})
 
 	t.Run("WithCorrectMFA", func(t *testing.T) {
-		t.Skip("Skipping test that fails randomly.")
-		code := dgoogauth.ComputeCode(secret.Secret, time.Now().UTC().Unix()/30)
+		secret3, err := th.App.GenerateMfaSecret(th.BasicUser.Id)
+		assert.Nil(t, err)
+
+		// Fake user has MFA enabled
+		if result := <-th.Server.Store.User().UpdateMfaActive(th.BasicUser.Id, true); result.Err != nil {
+			t.Fatal(result.Err)
+		}
+
+		if result := <-th.Server.Store.User().UpdateMfaSecret(th.BasicUser.Id, secret3.Secret); result.Err != nil {
+			t.Fatal(result.Err)
+		}
+
+		code := dgoogauth.ComputeCode(secret3.Secret, time.Now().UTC().Unix()/30)
 
 		user, resp := th.Client.LoginWithMFA(th.BasicUser.Email, th.BasicUser.Password, strconv.Itoa(code))
```

Comment:
@DSchalla, I think I've uncovered the core issue. dgoogauth's `ComputeCode` returns `int`, but it's Authenticate expects a `string`, and then validates the length as 6 (or 8 characters). After tracing the code to verify, I observed that this test fails whenever the string representation of the password is supposed to have a leading `zero`, but our `strconv.Itoa(code)` in the test discards same.

The library's own test code had exactly the same issue: https://github.com/dgryski/dgoogauth/commit/f176097db5de17a6c0f2f0d74abf9d38340fdd60.

As such we, can fix this test as follows:
```suggestion
		user, resp := th.Client.LoginWithMFA(th.BasicUser.Email, th.BasicUser.Password, fmt.Sprintf("%06d", code))
```

I'm unable to reproduce the issue with the changes above.

Reply:
:tada: Amazing find! This explains why I was never able to trigger it, considering the chance of a leading zero is not *that* high :-) Thanks a lot for the proposed fix and digging that deep, happy we got that fixed and can test the 2FA flow now properly.